src/main/java/software/bernie/techarium/material/armor/TechariumArmorMaterial.java

```diff
@@ -0,0 +1,67 @@
+package software.bernie.techarium.material.armor;
+
+import net.minecraft.inventory.EquipmentSlotType;
+import net.minecraft.item.IArmorMaterial;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.LazyValue;
+import net.minecraft.util.SoundEvent;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+import java.util.function.Supplier;
+
+public class TechariumArmorMaterial implements IArmorMaterial {
+    protected static final int[] MAX_DAMAGE_ARRAY = new int[]{13, 15, 16, 11};
+    protected final String name;
+    protected final int maxDamageFactor;
+    protected final int[] damageReductionAmountArray;
+    protected final int enchantability;
+    protected final SoundEvent soundEvent;
+    protected final float toughness;
+    protected final float knockbackResistance;
+    protected final LazyValue<Ingredient> repairMaterial;
+
+    public TechariumArmorMaterial(String nameIn, int maxDamageFactorIn, int[] damageReductionAmountsIn, int enchantabilityIn, SoundEvent equipSoundIn, float toughnessIn, float knockbackResistanceIn, Supplier<Ingredient> repairMaterialSupplier) {
+        this.name = nameIn;
+        this.maxDamageFactor = maxDamageFactorIn;
+        this.damageReductionAmountArray = damageReductionAmountsIn;
+        this.enchantability = enchantabilityIn;
+        this.soundEvent = equipSoundIn;
+        this.toughness = toughnessIn;
+        this.knockbackResistance = knockbackResistanceIn;
+        this.repairMaterial = new LazyValue(repairMaterialSupplier);
+    }
+
+    public int getDurabilityForSlot(EquipmentSlotType slotIn) {
+        return MAX_DAMAGE_ARRAY[slotIn.getIndex()] * this.maxDamageFactor;
+    }
+
+    public int getDefenseForSlot(EquipmentSlotType slotIn) {
+        return this.damageReductionAmountArray[slotIn.getIndex()];
+    }
+
+    public int getEnchantmentValue() {
+        return this.enchantability;
+    }
+
+    public SoundEvent getEquipSound() {
+        return this.soundEvent;
+    }
+
+    public Ingredient getRepairIngredient() {
+        return (Ingredient) this.repairMaterial.get();
+    }
+
+    @OnlyIn(Dist.CLIENT)
```

Comment:
```suggestion

```
Remove OnlyIn. This method is safe to call on server.
`This is generally meant for internal Forge and FML use only and modders should avoid its use whenever possible.`-JavaDoc of OnlyIn